beacon-chain/state/getters.go

```diff
@@ -450,13 +450,19 @@ func (b *BeaconState) ValidatorAtIndexReadOnly(idx uint64) (*ReadOnlyValidator,
 
 // ValidatorIndexByPubkey returns a given validator by its 48-byte public key.
 func (b *BeaconState) ValidatorIndexByPubkey(key [48]byte) (uint64, bool) {
+	if b == nil || b.valIdxMap == nil {
+		return 0, false
+	}
 	b.lock.RLock()
 	defer b.lock.RUnlock()
 	idx, ok := b.valIdxMap[key]
 	return idx, ok
 }
 
 func (b *BeaconState) validatorIndexMap() map[[48]byte]uint64 {
+	if b == nil || b.valIdxMap == nil {
```

Comment:
this is a bit dangerous. The caller of this method will get a nil map, which will definitely lead to a panic when they try to read from that map.

Reply:
I see, so short of rewriting this method to return error (or book ok), what can we do?

Reply:
Should I return inited empty map then?

Reply:
I think an empty map should be fine here instead

Reply:
Done! Thank you for noticing this.